### DISCUSSION TITLE
2D 360 Character rigging

### DISCUSSION THREAD

**Discussion by Mondaley** [opentoonz/opentoonz#6554] [Ideas]
### Describe the solution you'd like

I have reached a point in which I need more stable and time-saving tools to animate my characters in OpenToonz.
Drawing by hand each frame in between is awful and slow. Thus, having a 360-character ring system would be very helpful.

The already existing tools don't help in this matter:
 - The auto in-between from the level strip requires all lines to be in the same layer(otherwise they get tweened with the wrong ones) and all the vertices to be in roughly the same number and relative position from the start frame till the end (that means more work by tweaking the vertices manually). Also, the calculated frames don't respect the volume and the shape of the drawing most of the time, even if you respect those requirements. So it is useless in almost all cases.
![image](https://github.com/opentoonz/opentoonz/assets/122384675/f2f9ffeb-cbc1-4641-ad5c-62cac632d00e)
- The vector-guided drawing controls window eliminates the stroke layer requirement by letting you choose the two lines by yourself, but has its problems too: it doesn't respect the volume and the shape of the drawing and needs both lines to have the same flow(the direction the guiding arrows are pointing. You can change it via the settings but it is an unnecessary hassle to have in a such an app)
![image](https://github.com/opentoonz/opentoonz/assets/122384675/4e391296-3965-46f6-9d8d-07a08fdcadf1)
- The Range option from the brush tool has all the above issues plus it doesn't let you tweak the line in any way. So if you want a perfect stroke good luck. I don't know why this option even exists.
![image](https://github.com/opentoonz/opentoonz/assets/122384675/69393220-e438-4e99-aad6-ba2e3a8fcda2)
- The function editor Expression Interpolation is the closest thing I could find that lets me create 2d-360-rigg-like results, but this method is inefficient because it is not meant do to such tusks. That means it has a lot of limitations and gives poor results.

https://github.com/opentoonz/opentoonz/assets/122384675/48135c39-a0f4-48c3-8596-a81909d6c46c

![image](https://github.com/opentoonz/opentoonz/assets/122384675/203a8a0a-d093-45cd-9691-6fdda7dbe892)

- The bone tool make only 2D puppets but no 2D 360 Riggs
- and so on

I need something like this:

https://www.youtube.com/watch?v=8Utk_9sgjDM&ab_channel=StephenIosun

https://www.youtube.com/watch?v=-NpjgylgAcA&t=55s&ab_channel=Reallusion

Which is absent from the current software.

**Comment by beeheemooth** [comment]
Just wondering how long have you been using OT?

**Comment by Mondaley** [comment]
> Just wondering how long have you been using OT?

A couple of years :D

**Comment by Mondaley** [comment]
> If your computer can handle it, you might want to check out Blender and it's grease pencil capabilities. An animator friend of mine switched to it completely, while having previously used shockwave/adobe flash.
> 
> Since Blender is a 3d program, and grease pencil allows one to draw lines in 3d space, you will get all 3d space rotations for free.
> 
> https://www.youtube.com/watch?v=c57qq2nE3B0
> 
> And as usual, my disclaimer: I am suggesting different applications to OT sometimes to perhaps increase the chance for you to get your creativity out into the world, rather than having to wait for OT.

Yeah, Blender is better at 3D stuff.

I'll still use OT for designing though.

**Comment by RodneyBaker** [comment]
Transferring this over to Opentoonz-Docs
